cmd/shortener/handler/handler.go

```diff
@@ -0,0 +1,248 @@
+		sugarLogger.Infoln("request method", r.Method)
+
+		startRequestTime := time.Now()
+		handler.ServeHTTP(&responseHandler, r)
+		sugarLogger.Infoln("duration:", time.Since(startRequestTime))
+		sugarLogger.Infoln("\n----------------------------------------------------------\n\n")
+	})
+}
+
+type gzipWriter struct {
+	http.ResponseWriter
+	Writer *gzip.Writer
+}
+
+func newGzipWriter(w http.ResponseWriter) *gzipWriter {
+	sugarLogger.Infoln("new gzipWriter")
+	return &gzipWriter{
+		ResponseWriter: w,
+		Writer:         gzip.NewWriter(w),
+	}
+}
+
+func (w *gzipWriter) Write(b []byte) (int, error) {
+	sugarLogger.Infoln("response Content-Type", w.Header().Get("Content-Type"))
+	if !(strings.Contains(w.Header().Get("Content-Type"), "application/json") ||
+		strings.Contains(w.Header().Get("Content-Type"), "text/html")) {
+		sugarLogger.Infoln("call ResponseWriter:", string(b))
+		return w.ResponseWriter.Write(b)
+	}
+	sugarLogger.Infoln("call gzip write:", string(b))
+	defer w.Writer.Close()
+	return w.Writer.Write(b)
+}
+
+func GzipHandler(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		newHandler := w
+		if strings.Contains(r.Header.Get("Accept-Encoding"), "gzip") {
+			sugarLogger.Infoln("Accept-Encoding contains gzip:", r.Header.Get("Accept-Encoding"))
+			newHandler = newGzipWriter(w)
+			// defer newHandler.(*gzipWriter).Writer.Close()
+		}
+
+		if strings.Contains(r.Header.Get("Content-Encoding"), "gzip") {
+			sugarLogger.Infoln("request Content-Encoding", r.Header.Get("Content-Encoding"))
+			gz, err := gzip.NewReader(r.Body)
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+			defer gz.Close()
+			r.Body = gz
+		}
+
+		next.ServeHTTP(newHandler, r)
+		//next(w, r)
+	})
+}
+
+func GetShort(w http.ResponseWriter, r *http.Request) {
+	sugarLogger.Info("START GetShort")
+	shortURL := chi.URLParam(r, "short")
+	// println("shorturl", shortURL, len(urls), urls[shortURL])
+	url, ok := urlStore.Read(regexp.MustCompile(`[^a-zA-Z0-9 ]+`).ReplaceAllString(shortURL, ""))
+	sugarLogger.Infoln("original url from store to header.Location", url)
+	w.Header().Add("content-type", "text/plain")
+	if ok {
+		w.Header().Add("Location", url)
+		w.WriteHeader(http.StatusTemporaryRedirect)
+	} else {
+		w.WriteHeader(http.StatusBadRequest)
+	}
+}
+
+func PostURL(w http.ResponseWriter, r *http.Request) {
+	sugarLogger.Info("START PostURL")
+	body, bodyErr := io.ReadAll(r.Body)
+	sugarLogger.Infoln("get body", string(body))
+	if bodyErr != nil {
+		handleError(bodyErr, w)
+		return
+	}
+	w.Header().Set("Content-Type", "text/plain")
+	w.WriteHeader(http.StatusCreated)
+	shortURL, addr := shortName(body)
+	sugarLogger.Infoln("shortURL", shortURL)
+	sugarLogger.Infoln("addr", addr)
+	if writeToFile(shortURL, body) {
+		return
+	}
+	w.Write([]byte(addr + shortURL))
+}
+
+func writeToFile(shortURL string, body []byte) bool {
```

Comment:
```suggestion
func writeToFile(shortURL string, body []byte) error {
```
Если в процессе работы функции родилась ошибка, ее и надо возвращать. То, что результат записи в файл в виде `true` - это не записанный файл, вообще не очевидно.

Reply:
исправлено